packages/clay-shared/src/__tests__/FocusScope.tsx

```diff
@@ -298,15 +297,31 @@ describe('FocusScope', () => {
 				'button1'
 			) as HTMLElement;
 
-			dropdownButton.click();
+			fireEvent.click(dropdownButton);
+
+			const itemOne = (document.body as HTMLElement).querySelector(
+				'a[href="#one"]'
+			);
 
 			userEvent.tab();
 
 			expect(dropdownButton).toHaveFocus();
 
+			const itemTwo = (document.body as HTMLElement).querySelector(
+				'a[href="#two"]'
+			);
+
 			userEvent.tab();
 
 			expect(htmlButton1).toHaveFocus();
```

Comment:
Hey @matuzalemsteles! I've tried using the `FocusScope`  in the Dropdown to maintain the order as you said but it doesn't work. The order passes to the `button1` and continues to the links as the recent commit

Reply:
@sergiojimcos I did a quick review on this, I think this is an unexpected edge case, I'll try to explain what's going on.

Well, as I said we have to add the `FocusScope` as a wrapper of the elements that we want to control the focus, otherwise it will not work, because who does all the control of the focus is this component, the use of `ClayPortal` it changes where the element will be rendered in the DOM, this is to solve problems of dropdown menus that can be cut when some element with `overflow: hidden` exists inside a container, this is a strategy we did but the problem that this breaks the tab because the tab follows the declaration order of elements in the DOM, so we create `FocusScope` to manage the focus i.e. keep the tab order correctly, for example, consider this declaration order:

```jsx
<div>
  <button>Button 1</button>
  <div>
    <button>Button 2</button>
  </div>
  <button>Button 3</button>
</div>
```

This is a very simple case but just to illustrate, the tab order will follow the declaration order, so if you press the tab sequentially the focus order will be, `Button 1 -> Button 2 -> Button 3`.

The problem is that often the order of declaration may not make sense to the user, because the visual positioning may be different depending on the layout of the page, in scenarios where we use `ClayPortal` which breaks with the order of declaration, for example:

```jsx
<div>
  <button>Button 1</button>
  <ClayPortal>
    <button>Button 2</button>
  </ClayPortal>
  <button>Button 3</button>
</div>
```

This represented in the DOM is different:

```html
<body>
  <div id="root">
    <div>
      <button>Button 1</button>
      <button>Button 3</button>
    </div>
  </div>

  <button>Button 2</button>
</body>
```

So if there is no focus manager when pressing the tab sequentially it will be `Button 1 -> Button 3 -> Button 2`, then this is not intuitive. Then the focus manager will correct and maintain the tab order correctly. The scenario from this test case is a little different, as DXP is not an application completely built on top of React, React is used more as loose components and only the products are the application of the page this makes in many scenarios the ` FocusScope` will not act or be able to control the tab order, it only controls what is declared as `children` ie it listens for any tab event if it was pressed within the scope.

To better understand what is happening here, following this test markup, the representation in the DOM will be:

```html
<body>
  <div id="root">
    <div>
        <div class="dropdown dropdown-test">
          <button class="dropdown-toggle">Click Me</button>
        </div>
    </div>
  </div>

  <div id="content">
    <button id="button1"></button>
  </div>

  <div class="dropdown-menu">Menu content here</div>
</body>
```

The focus manager scope here is being the dropdown-toggle, button1 and the dropdown-menu, if you don't have a focus manager the sequential tab order here will be: `Click me -> button1 -> links in the dropdown menu` , we don't want this order because it breaks accessibility and will cause user confusion, so with the focus manager in theory the sequence order should be `button1 -> Click me -> links in the dropdown` the problem with this scenario is that as ClayPortal is rendered first ie it is the first element in the declaration order when the focus comes from outside the FocusScope has no way of detecting that a tab key was pressed inside the scope because the event goes to the window and we only listen for the event within the scope, this causes the focus order to be different than expected, `click me -> links in the dropdown -> button1`, because who controlled this focus was the browser that followed the declaration order in the DOM, this is the problem here because it's not following this behavior.

In short, this is an unexpected edge case because we don't expect the person to add a ClayPortal without some trigger, as this is internal to Clay we have full control over it, so basically every declared ClayPortal has some element before being declared in scope what makes it work well, for example, see this scenario.

```jsx
<FocusScope>
  <div>
    <button id="button1">Button 1</button>

    <ClayPortal>
      <div id="content">
        <button id="button2">Button 2</button>
      </div>
    </ClayPortal>

    <DropDownWithState>
      <ClayDropDown.ItemList>
        {[
          {href: '#one', label: 'one'},
          {href: '#two', label: 'two'},
        ].map((item, i) => (
          <ClayDropDown.Item
            href={item.href}
            key={i}
            spritemap="/foo/bar"
          >
            {item.label}
          </ClayDropDown.Item>
        ))}
      </ClayDropDown.ItemList>
    </DropDownWithState>
  </div>
</FocusScope>
```

This scenario will work as expected, `Button 1 -> Button 2 -> DropDown toggle -> links in the dropdown` this is because when the focus arrives within the scope now who is managing the focus is our component. I hope this is clearer, I'm going to send a commit to switch to this scenario. Another important thing about this is that this component also controls the reverse order as well.